sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceivedMessage.java

```diff
@@ -464,6 +445,25 @@ public String getSessionId() {
         return getRawAmqpMessage().getProperties().getGroupId();
     }
 
+    /**
+     * Gets the state of the message.
+     *
+     * The state of the message can be Active, Deferred, or Scheduled. Deferred messages have Deferred state, scheduled
+     * messages have Scheduled state, all other messages have Active state.
+     *
+     * @return The state of the message.
+     * @throws UnsupportedOperationException if the message state is an unknown value.
```

Comment:
I don't see this exception being thrown. Should the `else` part of the code be throwing instead of defaulting to `ACTIVE` then?

Reply:
Probably here? https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/ServiceBusMessageState.java#L58-L59